client/finality-grandpa/src/warp_proof.rs

```diff
@@ -192,37 +219,100 @@ impl<Block: BlockT> WarpSyncProof<Block> {
 			proof
 				.justification
 				.verify(current_set_id, &current_authorities)
-				.map_err(|err| HandleRequestError::InvalidProof(err.to_string()))?;
+				.map_err(|err| Error::InvalidProof(err.to_string()))?;
 
 			if proof.justification.target().1 != proof.header.hash() {
-				return Err(HandleRequestError::InvalidProof(
-					"mismatch between header and justification".to_owned()
+				return Err(Error::InvalidProof(
+					"Mismatch between header and justification".to_owned()
 				));
 			}
 
 			if let Some(scheduled_change) = find_scheduled_change::<Block>(&proof.header) {
 				current_authorities = scheduled_change.next_authorities;
 				current_set_id += 1;
-			} else if fragment_num != self.proofs.len() - 1 {
-				// Only the last fragment of the proof is allowed to be missing the authority
+			} else if fragment_num != self.proofs.len() - 1 || !self.is_finished {
```

Comment:
@andresilva Please check if this is correct.

Reply:
Seems correct to me, only the last block in the last fragment can be for a block that doesn't change the authority set.